src/rabbit_mgmt_stats_gc.erl

```diff
@@ -134,15 +138,50 @@ gc_batch(Rows, Policies, State = #state{gc_next_key = Cont,
                       Now = floor(
                               time_compat:os_system_time(milli_seconds),
                               State),
-                      gc(Key, Table, Policies, Now),
+                      gc(Key, Table, Config, Now),
                       Key
               end,
-    gc_batch(Rows - 1, Policies, State#state{gc_next_key = NewCont}).
+    gc_batch(Rows - 1, Config, State#state{gc_next_key = NewCont}).
 
-gc(Key, Table, Policies, Now) ->
-    Policy = pget(retention_policy(Table), Policies),
+gc(Key, Table, Config, Now) ->
+    case lists:member(Table, ?PROC_STATS_TABLES) of
```

Comment:
They should be almost the same, it might even be possible that the compiler changes them into the same instructions.
In fact, it will be easier (in dev and performance terms) if you simply create a new module for the new GC. There might be a little bit of code duplication, but it is a small module and I don't see any problem on that.